src/rageshake/rageshake.js

```diff
@@ -241,20 +241,27 @@ class IndexedDBLogStore {
 
         // Returns: a string representing the concatenated logs for this ID.
         function fetchLogs(id) {
-            const o = db.transaction("logs", "readonly").objectStore("logs");
-            return selectQuery(o.index("id"), IDBKeyRange.only(id),
-            (cursor) => {
-                return {
-                    lines: cursor.value.lines,
-                    index: cursor.value.index,
+            const objectStore = db.transaction("logs", "readonly").objectStore("logs");
+
+            return new Promise((resolve, reject) => {
+                const query = objectStore.index("id").openCursor(IDBKeyRange.only(id), 'next');
+                let lines = '';
+                query.onerror = (event) => {
+                    reject(new Error("Query failed: " + event.target.errorCode));
+                };
+                query.onsuccess = (event) => {
+                    const cursor = event.target.result;
+                    if (!cursor) {
+                        resolve(lines);
+                        return; // end of results
+                    }
+                    lines += cursor.value.lines;
```

Comment:
this includes a newline already?

Reply:
yeah, or at least it was previously just `lines.join('')`...